src/orng.cpp

```diff
@@ -490,7 +515,17 @@ void orng::setrand(name oracle, uint64_t id, uint8_t ver, std::string sig){
         std::tuple(rit->assoc_id, rnd)
     ).send();    
 
+    if(rit->free_call){
```

Comment:
@hlongvu why don't we do this logic in the requestrand function where we do the free call logic below the treas.pool_balance check? I think we could run the risk of overrunning the treas.pool_balance if we queue too many jobs between that check and here.